roles/bootstrap-os/tasks/main.yml

```diff
@@ -55,6 +55,7 @@
 - name: Assign inventory name to unconfigured hostnames (non-CoreOS, non-Flatcar, Suse and ClearLinux, non-Fedora)
   hostname:
     name: "{{ inventory_hostname }}"
+    use: '{{ "systemd" if ''ID="rocky"'' in os_release.stdout_lines else omit }}'
```

Comment:
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/hostname_module.html says 
```
Note that 'systemd' should be specified for RHEL/EL/CentOS 7+.
```
Why here specifies `systemd` only if rocky linux?

Reply:
My use case was about Rocky Linux, but I guess you’re right, we better be specifying `systemd` for RHEL-based 7+ OS. However, I don’t know how should I do that?

Should I use the `ansible_distribution_major_version`? On Rocky and Alma its value is "8". I’m going to try on CentOS but can’t test on RedHat (although it should be the same as CentOS…).

What do you think about that?

Reply:
I think it would make more sense to detect `systemd` based distributions and set `use: systemd` when a running systemd is detected, that way we could simplify the detection logic a bit.

Reply:
> I’m going to try on CentOS but can’t test on RedHat (although it should be the same as CentOS…).

We are testing CentOS7, CentOS8 on Kubespray CI.
It is fine to use the CI without manual testing on your side for CentOS.
Now @cristicalin is trying to replace CentOS8 CI with AlmaLinux with https://github.com/kubernetes-sigs/kubespray/pull/8297
So we will not test on Rocky Linux, it is very nice to test on Rocky Linux on your side.

/ok-to-test

Reply:
My plan is to modify #8297 after we tag 2.18 and spread the jobs between Alma and Rocky. One thing to note is that this fix is only needed for ansible < 2.11. With ansible 2.11 onward we have full support for Rocky and the `hostname` module correctly recognizes the distribution.

I was thinking we could use this change to simplify our OS detection logic and avoid checking specifically for Alma, Rocky, CentOS and other derivatives.

Reply:
> My plan is to modify #8297 after we tag 2.18 and spread the jobs between Alma and Rocky. One thing to note is that this fix is only needed for ansible < 2.11. With ansible 2.11 onward we have full support for Rocky and the `hostname` module correctly recognizes the distribution.
> 
> I was thinking we could use this change to simplify our OS detection logic and avoid checking specifically for Alma, Rocky, CentOS and other derivatives.

That is a nice plan to improve the test coverage for distributions, thanks @cristicalin !

Reply:
This is very nice to hear @cristicalin ! I didn’t know `hostname` module worked as expected on Rocky with ansible 2.11+.

So, from my side, should I try to make the check aware of all RHEL 7+ based OS or is it ok to leave it as-is?

In order for kubespray to work with Rocky, this is needed as well as overriding `ansible_os_family` to set it to `RedHat`. What do you want me to do then?

Reply:
If your use-case allows for using ansible 2.11 I would suggest we don't perform this change that is too specific to Rocky and instead try to apply a broader approach when setting `use: systemd`.

With regard to overriding `ansible_os_family` this looks like an ugly hack that would be better not to land in kubespray.

Reply:
> With regard to overriding ansible_os_family this looks like an ugly hack that would be better not to land in kubespray.

I agree. That would make Kubespray code hard to be maintained in long-term.

Reply:
I also agree, and that’s why I keep this hack as an extra variable when I run kubespray.

As per the change introduced in this PR, I’d have to try running kubespray with ansible 2.11 but as far as I remember I’d already tried that and got stuck by the ansible version check. As I can see the `maximal_ansible_version` has been increased to 2.12 so this should work. Will try again tomorrow and let you know.

Reply:
I can confirm `hostname` module works as expected with Rocky Linux when using ansible 2.11 (through `requirements-2.11.txt`)! I also don’t need to specify the extra variable `ansible_os_family` anymore :partying_face: 

Therefore I’m closing this PR.

Thanks for your help/insights!